### PR TITLE
[CCP] initial state and filter

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
Makes the filter options tied to data from the patient record. Previously these values were hard coded.  Some of the data is missing from the patient record, and the filter categories aren't final, or rather are supposed to be generalized but I'm not sure where that information is supposed to come from or what it looks like.  So this is still a bit WIP.

There is still outstanding work to do on what happens when the data is not available from the patient record. Dan has suggested this be a dropdown input the physician can choose a value from. If we do it this way, that should probably be a separate JIRA.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by Dtphelan1** [comment]
[DP1] Bunch of warnings that need addressing:
image

File: src/mcode-pilot/components/SimilarPatientsSelector/SimilarPatientsSelector.jsx [hunk 1]
```diff
@@ -31,12 +31,17 @@ export default class SimilarPatientsSelector extends Component {
...
                 <div className="similar-patients-selector__options">
-                    <OptionsCheckboxList
-                        options={similarPatientProps.demographic}
-                        setSelected={(key, selected) => selectSimilarPatientOption('demographic', key, selected)}
-                        setAllSelected={(selected) => selectAllCategorySimilarPatientOptions('demographic', selected)}
-                    />
+                    { Object.keys(similarPatientProps).map((option)=>{
+                        return (
+                            <OptionsCheckboxList
+                            options={similarPatientProps[option]}
```
Comment (Dtphelan1) [inline src/mcode-pilot/components/SimilarPatientsSelector/SimilarPatientsSelector.jsx]: Could we indent these props to align with styles across the app?

File: src/mcode-pilot/components/SimilarPatientsSelector/SimilarPatientsSelector.jsx [hunk 2]
```diff
@@ -31,12 +31,17 @@ export default class SimilarPatientsSelector extends Component {
...
+                            setSelected={(key, selected) => selectSimilarPatientOption(option, key, selected)}
+                            setAllSelected={(selected) => selectAllCategorySimilarPatientOptions(option, selected)}
+                            />
+                        )
+                    })}
 
+{/*
```
Comment (Dtphelan1) [inline src/mcode-pilot/components/SimilarPatientsSelector/SimilarPatientsSelector.jsx]: Let's remove this commented out code

**Comment by Dtphelan1** [comment]
@KeeyanGhoreshi let us know when this is rebased onto Master now that Noranda's PR is merged.

**Comment by noranda** [comment]
I believe in the case where we don't have the patient info, we want to exclude that option, at least for the MVP. Is that something you can include in this PR relatively easily? If not, we should make a new task for it. Otherwise this looks good to me.

**Comment by noranda** [comment]
Also needs rebasing again.

**Comment by mgramigna** [comment]
This looks good to me. I just have a few code style things that I'll make as separate comments

File: src/mcode-pilot/components/SimilarPatientsSelector/SimilarPatientsSelector.jsx [hunk 3]
```diff
@@ -31,35 +31,16 @@ export default class SimilarPatientsSelector extends Component {
...
-                        setSelected={(key, selected) => selectSimilarPatientOption('genetics', key, selected)}
-                        setAllSelected={(selected) => selectAllCategorySimilarPatientOptions('genetics', selected)}
-                    />
-
-                    <OptionsCheckboxList
-                        options={similarPatientProps.medicalHistory}
-                        setSelected={(key, selected) => selectSimilarPatientOption('medicalHistory', key, selected)}
-                        setAllSelected={(selected) => selectAllCategorySimilarPatientOptions('medicalHistory', selected)}
-                    />
+                    { Object.keys(similarPatientProps).map((option)=>{
```
Comment (mgramigna) [inline src/mcode-pilot/components/SimilarPatientsSelector/SimilarPatientsSelector.jsx]: Space after parentheses and before bracket

File: src/mcode-pilot/utils/recordToProps.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
+import FluxTumorDimensions from '../../dataaccess/mcodev0.1-datasource/model/oncology/FluxTumorDimensions';
+
+import _ from 'lodash';
+
+export default function getProps(patient, condition){
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Space before bracket

File: src/mcode-pilot/utils/recordToProps.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                    "range":1,
+                    "value":safeGet(condition.getMostRecentECOGPerformanceStatus(),"value")
+
+                }
+        }
+     
+        
+    }
+
+    if(tumorMarkers){
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after if and before bracket.

Spaces after parentheses and before bracket

File: src/mcode-pilot/utils/recordToProps.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                "value": _.lowerCase(e.status)
+            }
+        });
+    }
+
+    return mapProp(propDict);
+}
+
+function safeGet(object, property) {
+    if(object !== null & object!==undefined && property in object){
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after if and before bracket.

File: src/mcode-pilot/utils/recordToProps.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        return object[property]
+    }else{
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after bracket and before bracket.

File: src/mcode-pilot/utils/recordToProps.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    const similarPatientProps = {}
+    // categories
+    for (const key of Object.keys(propDict)) {
+        similarPatientProps[key] = {
+            options:{},
+            selected:false,
+            displayText: key
+        }
+        // values
+        for (const prop of Object.keys(propDict[key])){
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: space before bracket

File: src/mcode-pilot/utils/recordToProps.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            const option = propDict[key][prop];
+            // drops option boxes that don't have 
+            // a value from the patient record
+            if(!option.value){
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after if and before bracket.

File: src/mcode-pilot/utils/recordToProps.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+                let propEntry = {
+                    selected: false,
+                    displayText: option.display,
+                    value:"-"
+                };
+                similarPatientProps[key].options[prop] = propEntry;
+            }
+            if(option.value){
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after if and before bracket.

File: src/mcode-pilot/utils/recordToProps.js [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+                let propEntry = {
+                    selected: false,
+                    displayText: option.display
+                };
+                if(option.valueType==="range") {
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after if and before bracket.

File: src/mcode-pilot/utils/recordToProps.js [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+                    propEntry.minValue = (option.value >= option.range)?option.value-option.range:0;
+                    propEntry.maxValue = option.value + option.range
+                }else{
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: Spaces after bracket and before bracket.

File: src/mcode-pilot/utils/recordToProps.js [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+                    propEntry.value = option.value;
+                }
+                similarPatientProps[key].options[prop] = propEntry;
+            }
+
```
Comment (mgramigna) [inline src/mcode-pilot/utils/recordToProps.js]: remove blank line

File: src/reducers/mcode.js [hunk 14]
```diff
@@ -4,32 +4,11 @@ import _ from 'lodash';
 import filterTreatmentData from '../mcode-pilot/utils/filterTreatmentData';
 
 import defaultState from './initial.json';
-
+import getProps from '../mcode-pilot/utils/recordToProps'
```
Comment (mgramigna) [inline src/reducers/mcode.js]: semicolon

File: test/backend/mcode/TreatmentOptionsReducers.test.js [hunk 15]
```diff
@@ -1,51 +1,55 @@
...
-            }
-            const newState = reducer(undefined,action).similarPatientProps.demographic.options;
-            expect(newState.race.value).to.eql("sprint");
-            expect(newState.gender.value).to.eql("male")
-            expect(newState.age.minValue).to.eql(0);
-            expect(newState.age.maxValue).to.eql(action.patientAge+10);
-            expect(newState.diagnosedAge.minValue).to.equal(0);
-            expect(newState.diagnosedAge.maxValue).to.equal(action.patientAgeAtDiagnosis+10);
         });
+
```
Comment (mgramigna) [inline test/backend/mcode/TreatmentOptionsReducers.test.js]: remove blank line

**Comment by KeeyanGhoreshi** [comment]
> I believe in the case where we don't have the patient info, we want to exclude that option, at least for the MVP. Is that something you can include in this PR relatively easily? If not, we should make a new task for it. Otherwise this looks good to me.

This has been added.  This will also exclude the entire section if there are no options present (for example, `medical history` is excluded if `ECOG Score` is excluded.

**Comment by noranda** [comment]
Once this is rebased, I will approve.

**Comment by KeeyanGhoreshi** [comment]
should be rebased now, I thought I rebased it before but I was rebasing on my out of date local version of master.  Should be good now.

**Comment by noranda** [comment]
If you can rebase this one more time, I'll approve and merge!

**Pull request merged**